spec/fabricators/web_push_subscription_fabricator.rb

```diff
@@ -2,6 +2,10 @@
 
 Fabricator(:web_push_subscription, from: Web::PushSubscription) do
   endpoint   Faker::Internet.url
-  key_p256dh Faker::Internet.password
-  key_auth   Faker::Internet.password
+  key_p256dh do
+    curve = OpenSSL::PKey::EC.generate('prime256v1')
+    ecdh_key = curve.public_key.to_bn.to_s(2)
+    Base64.urlsafe_encode64(ecdh_key)
```

Comment:
@ClearlyClaire I was using the same code as in the [WebPush library](https://github.com/ClearlyClaire/webpush/blob/master/spec/webpush/encryption_spec.rb#L5-L10), however, found that the OpenSSL interface has changed, resulting in the following error
```
Failure/Error: curve.generate_key

      OpenSSL::PKey::PKeyError:
        pkeys are immutable on OpenSSL 3.0
```

We may want to make sure we're testing the WebPush library against the supported ruby versions, particular 3.1 and above.

Reply:
This migration has long been done, it's been at least a year that we primarily run everything on OpenSSL 3.0 and recent ruby versions.

Regarding my WebPush library fork, you want to check the `openssl-3` branch: https://github.com/ClearlyClaire/webpush/blob/openssl-3/spec/webpush/encryption_spec.rb#L5-L10

Reply:
Ah cool!